relayer/relay/src/chain.rs

```diff
@@ -35,6 +38,19 @@ pub trait Chain {
     /// Type of RPC requester (wrapper around low-level RPC client) for this chain
     type Requester: tmlite::Requester<Self::Commit, Self::Header>;
 
+    /// Error types defined by this chain
+    type Error: Into<Box<dyn Error + Send + Sync + 'static>>;
+
+    /// Perform a generic `query`, and return the corresponding deserialized response data.
+    // This is going to be a blocking request.
```

Comment:
Not very important right now, but just pointing out that async methods are somewhat supported on stable Rust via the [`async-trait`](https://docs.rs/async-trait) crate. This crate sometimes yield weird compile-time errors and incurs a [boxing of the returned `Future`](https://docs.rs/async-trait/0.1.36/async_trait/#explanation) but has otherwise worked pretty well for us in the past.

Reply:
Cool! That's the kind of thing I need. I'll take this info away and use it for implementations for subscription and/or transactions.
I want to stop "improving" queries and if this stays as a non-async implementation for ow, it still works out fine.
Thanks for the info and thanks for understanding!